tests/monitoring_test.go

```diff
@@ -0,0 +1,228 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	openshiftroutev1 "github.com/openshift/api/route/v1"
+	promApi "github.com/prometheus/client_golang/api"
+	promApiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	promConfig "github.com/prometheus/common/config"
+	apps "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands/metrics"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+var promClient promApiv1.API
+
+var _ = Describe("Prometheus Alerts", func() {
+	BeforeEach(func() {
+		waitUntilDeployed()
+		promClient = initializePromClient(getPrometheusUrl(*coreClient), getAuthorizationTokenForPrometheus(*coreClient))
+	})
+
+	Context("SSPCommonTemplatesModificationReverted", func() {
+		var (
+			testTemplate testResource
+		)
+		BeforeEach(func() {
+			strategy.SkipIfUpgradeLane()
+			testTemplate = createTestTemplate()
+		})
+		It("[test_id:???] Should fire SSPCommonTemplatesModificationReverted", func() {
+			// we have to wait for prometheus to pick up the series before we increse it.
+			waitForSeriesToBeDetected("sum(increase(total_restored_common_templates{pod=~'ssp-operator.*'}[1h]))")
+			expectTotalRestoredTemplatesCountToIncrease(testTemplate)
```

Comment:
From the name of this function it is not clear that the template is updated inside.